### PR TITLE
fix: Refactor agenda google connector - EXO-65042

### PR DESCRIPTION
**Pull request by hakermi**
Prior to this change, the secert key was needed to be able to use the google api, in the first implementation the seceret_key was exposed in front side  in js files which is considered as a security issue.
This PR changes the way to get the access and refresh tokens by implementing a sever side part which will be responsible about getting, refreshing and storing tokens

### CODE REVIEW

File: agenda-connectors-webapp/src/main/webapp/vue-app/agenda-connectors/google-connector/agendaGoogleConnector.js [hunk 1]
```diff
@@ -37,7 +36,6 @@ export default {
       throw new Error('Google connector can\'t be enabled with empty Client API Key or empty Client Secret Key.');
     }
     this.CLIENT_ID = apiKey;
-    this.SECRET_KEY = secretKey;
```
Comment (rdenarie) [inline agenda-connectors-webapp/src/main/webapp/vue-app/agenda-connectors/google-connector/agendaGoogleConnector.js:40]: secreKey should be removed from the call
In addition, we must ensure that, in the caller, we no more get the secretKey from server side

File: agenda-connectors-webapp/src/main/webapp/vue-app/agenda-connectors/google-connector/agendaGoogleConnector.js [hunk 2]
```diff
@@ -62,29 +60,29 @@ export default {
...
+            }
+          });
+        } else if (this.user) {
+          return getStoredToken().then(tokenResponse => {
+            if (tokenResponse && tokenResponse.access_token) {
+              this.gapi.client.setToken(tokenResponse);
+              this.canPush = this.cientOauth.hasGrantedAllScopes(tokenResponse, this.SCOPE_WRITE);
+              this.gapi.client.setToken(tokenResponse);
+              const g_stateCookie = getCookie('g_state');
+              setCookie('g_state', g_stateCookie, 90);
```
Comment (rdenarie) [inline agenda-connectors-webapp/src/main/webapp/vue-app/agenda-connectors/google-connector/agendaGoogleConnector.js]: Do we still need the g_state cookie ? What do we store into ?

Comment (hakermi) [reply agenda-connectors-webapp/src/main/webapp/vue-app/agenda-connectors/google-connector/agendaGoogleConnector.js]: removed

File: agenda-connectors-webapp/src/main/webapp/vue-app/agenda-connectors/google-connector/agendaGoogleConnector.js (hunk 1 above)
Comment (hakermi) [reply agenda-connectors-webapp/src/main/webapp/vue-app/agenda-connectors/google-connector/agendaGoogleConnector.js:40]: done here https://github.com/exoplatform/agenda/pull/557

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you update the commit message to explain the modification ?
In addition it is more a fix than a feat, so can you also update the commit title ?
thanks

**Comment by hakermi** [comment]
> Can you update the commit message to explain the modification ? In addition it is more a fix than a feat, so can you also update the commit title ? thanks

Done

**Review by rdenarie — APPROVED** [review]

**Pull request merged**